### ISSUE TITLE
sqlfluff fix for rule L036 does not follow comma_style

### ISSUE THREAD

**Issue by jessie-carta** [sqlfluff/sqlfluff#1312]
Running sqlfluff fix on rule L036 with comma_style = leading does not behave as expected.

## Steps to Reproduce
Run `sqlfluff fix --rules L036` on

```
SELECT a, b
FROM table
```

## Expected Behaviour

```
SELECT 
  a
  , b
FROM table
```

or
```
SELECT 
    a
  , b
FROM table
```

## Observed Behaviour

```
SELECT
a,
b
FROM table
```

## Dialect
postgres

## Version
sqlfluff, version 0.6.2
Python 3.7.11

## Configuration
```
[sqlfluff]
nocolor = True
dialect = postgres
templater = dbt
output_line_length = 120
ignore = parsing, templating

[sqlfluff:rules]
comma_style = leading
max_line_length = 120
tab_space_size = 2
```

**Comment by tunetheweb** [comment]
You need to include L019 to fix trailing commas at the same time. Try this:

```
sqlfluff fix --rules L036,L019
```

That works for me on your example SQL.

**Comment by zhongjiajie** [comment]
> You need to include L019 to fix trailing commas at the same time. Try this:
> 
> ```
> sqlfluff fix --rules L036,L019
> ```
> 
> That works for me on your example SQL.

Should we add FAQ like those usage?

**Comment by tunetheweb** [comment]
I don't believe we need to. When you explicitly say to only run rule L036 (add new lines to `SELECT` for multiple arguments), I don't think there's an expectation that other rules (e.g. L019 - enforce leading commas) will also run. Quote the opposite - I think there's an expectation they will not run!

In fact [the example for L036](https://docs.sqlfluff.com/en/stable/rules.html#sqlfluff.core.rules.Rule_L036) explicitly uses trailing commas: 

```sql
select
    a,
    b,
    c
from x
```

Can you explain more what you think should be added to where and why, in case I'm missing something?

**Comment by jessie-carta** [comment]
I was confused because I specified this in the config 
```
[sqlfluff:rules]
comma_style = leading
```
so I assumed any fixes would apply leading commas. It might be less confusing to have either just the L019 rule to define `comma_style` or just the `.sqlfluff` config to define it, rather than having both.

**Comment by tunetheweb** [comment]
That's a fair point. Since this config only applies to L019 (at least at this time) it probably doesn't need to be a global config. Not sure if it will ever be needed by another rule in the future. In some ways it's quite confusing that parameters can be defined at so many levels and usually don't need to be since they are unique names.

But I think the point still stands that when you are only running a subset of the rules (and only one in this case), you should not expect the other rules to run. For example `indent_unit` (another global config) is used by L003, L004 and L016 but just setting that config doesn't imply that those three rules will be run when you say only to run L036.
